PICMI_Python/applied_fields.py

```diff
@@ -0,0 +1,118 @@
+      - Bx_expression: Analytic expression describing Bx field (string) [T]
+      - By_expression: Analytic expression describing By field (string) [T]
+      - Bz_expression: Analytic expression describing Bz field (string) [T]
+               Expressions should be in terms of the position and time, written as 'x', 'y', 'z', 't'.
+               Parameters can be used in the expression with the values given as additional keyword arguments.
+               Expression should relative to the lab frame.
+      - lower_bound=[None,None,None]: Lower bound of the region where the field is applied (vector) [m]
+      - upper_bound=[None,None,None]: Upper bound of the region where the field is applied (vector) [m]
+    """
+    def __init__(self, Ex_expression=None, Ey_expression=None, Ez_expression=None,
+                       Bx_expression=None, By_expression=None, Bz_expression=None,
+                 lower_bound=[None,None,None], upper_bound=[None,None,None],
+                 **kw):
+
+        self.Ex_expression = Ex_expression
+        self.Ey_expression = Ey_expression
+        self.Ez_expression = Ez_expression
+        self.Bx_expression = Bx_expression
+        self.By_expression = By_expression
+        self.Bz_expression = Bz_expression
+
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+        # --- Find any user defined keywords in the kw dictionary.
+        # --- Save them and delete them from kw.
+        # --- It's up to the code to make sure that all parameters
+        # --- used in the expression are defined.
+        self.user_defined_kw = {}
+        for k in list(kw.keys()):
+            if ((self.Ex_expression is not None and re.search(r'\b%s\b'%k, self.Ex_expression)) or
+                (self.Ey_expression is not None and re.search(r'\b%s\b'%k, self.Ey_expression)) or
+                (self.Ez_expression is not None and re.search(r'\b%s\b'%k, self.Ez_expression)) or
+                (self.Bx_expression is not None and re.search(r'\b%s\b'%k, self.Bx_expression)) or
+                (self.By_expression is not None and re.search(r'\b%s\b'%k, self.By_expression)) or
+                (self.Bz_expression is not None and re.search(r'\b%s\b'%k, self.Bz_expression))):
+                self.user_defined_kw[k] = kw[k]
+                del kw[k]
+
+        self.handle_init(kw)
+
+
+class PICMI_Mirror(_ClassWithInit):
+    """
+    Describes a perfectly reflecting mirror, where the E and B fields are zeroed
+    out in a plane of finite thickness.
+      - x_front_location: Location in x of the front of the nirror (float) [m]
+      - y_front_location: Location in y of the front of the nirror (float) [m]
+      - z_front_location: Location in z of the front of the nirror (float) [m]
+      - depth: Depth of the mirror (float) [m]
+      - number_of_cells: Minimum numer of cells zeroed out (integer)
+
+    Only one of the [x,y,z]_front_location should be specified.
```

Comment:
If you wish to include a bit more of description, maybe you can add something like:
```suggestion
    Only one of the [x,y,z]_front_location should be specified and the mirror will be set perpendicular to the respective direction and infinite in the other ones.
```